Login page preceding content pages

washort opened the pull request.

Commit:
Login page preceding content pages

openjck commented:
Looks like tests are failing

spasovski commented on distributionviewer/core/static/js/login.js:
```diff
@@ -0,0 +1,20 @@
+    body: new URLSearchParams('token=' + user.token)});
+  window.fetch(req).then(function (response) {
+    console.log('signed in:', user.name);
+    window.location.pathname = document.children[0].dataset.logintarget;
```
I'd do `document.querySelector('html').dataset...` for readability.

spasovski commented on distributionviewer/core/static/js/login.js:
```diff
@@ -0,0 +1,20 @@
+    console.log('signed in:', user.name);
+    window.location.pathname = document.children[0].dataset.logintarget;
+  }).catch(function (err) {
+    console.log("Login error:", err)
```
Stick with `'` for consistency.

spasovski committed:
add sign out button

Commit:
Kill Django session on logout

Commit:
linter fixes

openjck committed:
Update logout button style

openjck committed:
Update login button style

openjck committed:
Default to redirecting to /

This fixes a bug where the user would be redirected to None if ?next was
not set.

openjck committed:
Use .button style for logout button

openjck commented:
Updated the styles of the login and logout buttons. Unfortunately we can't change the style of the login button too much. Quoting Google from [this article](https://developers.google.com/identity/sign-in/web/build-button#building_a_button_with_a_custom_graphic):

> You must follow the branding guidelines and use the appropriate colors and icons in your button.

Everything else gets an r+ from me.
